lib/rpmts.cc

```diff
@@ -406,6 +406,50 @@ rpmRC rpmtsImportPubkey(const rpmts ts, const unsigned char * pkt, size_t pktlen
     return rc;
 }
 
+rpmRC rpmtsRebuildKeystore(rpmtxn txn, const char * from)
+{
+    rpmts ts = rpmtxnTs(txn);
+    rpmRC rc = RPMRC_OK;
+    rpmKeyring keyring = rpmtsGetKeyring(ts, 1);
+    keystore_fs ks_fs = {};
+    keystore_rpmdb ks_rpmdb = {};
+    keystore_openpgp_cert_d ks_opengpg = {};
+    rpmKeyringIterator iter = NULL;
+
+    if (rpmtsOpenDB(txn->ts, (O_RDWR|O_CREAT))) {
```

Comment:
This looks just wrong. If we're rebuilding an fs or openpgp keyring, the rpmdb doesn't come into play at all.

Reply:
Fixed. There was quite a mess of part relying one the surrounding code opening the rpmdb and part in the keystore opening the rpmdb themselves. All code paths in the rpmdb key store should now open the rpmdb.